tests/net/mocks/assert.c

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include "assert.h"
+
+DEFINE_FAKE_VALUE_FUNC(bool, mock_check_if_assert_expected);
+
+void assert_print(const char *fmt, ...)
+{
+	va_list ap;
+
+	va_start(ap, fmt);
+	vprintk(fmt, ap);
+	va_end(ap);
+}
+
+void assert_post_action(const char *file, unsigned int line)
+{
+	/* ztest_test_pass()/ztest_test_fail() are used to stop the execution
+	 * If this is an unexpected assert (i.e. not following expect_assert())
+	 * calling mock_check_if_assert_expected() will return 'false' as
+	 * a default return value
+	 */
+	if (mock_check_if_assert_expected() == true) {
```

Comment:
```suggestion
	if (mock_check_if_assert_expected()) {
```

Reply:
That is the code I copied from `tests/bluetooth/host/host/mock/`, thus i.e. copyrights were left unchanged